From ddc619605157ddd3d99d41f54e305f9f3b3bd594 Mon Sep 17 00:00:00 2001
From: Varsha Maira <vmaira@habana.ai>
Date: Tue, 13 May 2025 22:18:23 +0300
Subject: [PATCH] [SW-218933] g3: rename access mode enum items

The mem_access_mode enum is currently defined in multiple apps —
preboot_rtos, preboot, and zephyr-plugins. To eliminate code duplication,
it will be moved to the specs folder.

This patch renames the existing enum items to avoid conflicts and ensure
CI passes when the enum is added to the specs folder in a subsequent patch

Change-Id: I8b72dc1e7cd4b043e2ee394420a847f94a116ec8
Signed-off-by: Varsha Maira <vmaira@habana.ai>
---

diff --git a/common/include/security.h b/common/include/security.h
index 35d1238..bafcc4f 100644
--- a/common/include/security.h
+++ b/common/include/security.h
@@ -16,9 +16,9 @@
 #define KEY_PROP_EXP_SZ		8
 
 enum fwloader_mem_access_mode {
-	ACCESS_RO,
-	ACCESS_WR,
-	ACCESS_RW,
+	MEM_ACCESS_RO,
+	MEM_ACCESS_WR,
+	MEM_ACCESS_RW,
 };
 
 /* This structure is suitable for any min/max based RRs */
diff --git a/common/src/gaudi3/security.c b/common/src/gaudi3/security.c
index 53440f1..0ef105d 100644
--- a/common/src/gaudi3/security.c
+++ b/common/src/gaudi3/security.c
@@ -130,7 +130,7 @@
 	 * RR5 is used to enforce the write protection on the SRAM region dynamically
 	 * opened to the host, as it is exposed with read-only (RO) permissions.
 	 */
-	if (mode == ACCESS_RO)
+	if (mode == MEM_ACCESS_RO)
 		hl_sec_config_dyn_rr(HL_SEC_HBW_RR_5, image_base_full, img_max, HL_SEC_WR_PROT);
 
 	/*
diff --git a/fw-loader/src/fw_upd.c b/fw-loader/src/fw_upd.c
index 29dce17..0f68f55 100644
--- a/fw-loader/src/fw_upd.c
+++ b/fw-loader/src/fw_upd.c
@@ -336,7 +336,7 @@
 		}
 
 		if (tmp_offset)
-			hl_sec_dyn_img_open((uintptr_t)tmp_offset, fwstate.res_size, ACCESS_RO);
+			hl_sec_dyn_img_open((uintptr_t)tmp_offset, fwstate.res_size, MEM_ACCESS_RO);
 	default:
 		/* on success the original offset is set,  */
 		fwupd_send_status_simple(sts, tmp_offset);
@@ -655,7 +655,7 @@
 		goto prog_timeout;
 
 	if (fwstate.res_size) {
-		hl_sec_dyn_img_open((uintptr_t)fwstate.buf, fwstate.res_size, ACCESS_RW);
+		hl_sec_dyn_img_open((uintptr_t)fwstate.buf, fwstate.res_size, MEM_ACCESS_RW);
 		fwupd_send_status(HL_UPD_STS_PROG_OK, fwstate.buf);
 	} else {
 		fwupd_send_status(HL_UPD_STS_PROG_OK, 0);
@@ -694,7 +694,7 @@
 
 	/* make sure the data is in RAM */
 	sys_cache_flush((void *)fwstate.buf, sz);
-	hl_sec_dyn_img_open((uintptr_t)fwstate.buf, sz, ACCESS_RW);
+	hl_sec_dyn_img_open((uintptr_t)fwstate.buf, sz, MEM_ACCESS_RW);
 
 	if (fwupd_wfc(WFC_TIMEOUT_MS))
 		goto map_timeout;
@@ -835,7 +835,7 @@
 		goto resp_end;
 	}
 
-	hl_sec_dyn_img_open((uintptr_t)fwstate.buf, fwstate.res_size, ACCESS_RO);
+	hl_sec_dyn_img_open((uintptr_t)fwstate.buf, fwstate.res_size, MEM_ACCESS_RO);
 	fwupd_send_status_simple(HL_UPD_STS_OK, fwstate.buf);
 
 	return CMD_RET_SUCCESS;
